napari/layers/base/base.py

```diff
@@ -906,6 +906,24 @@ def _slice_dims(self, point=None, ndisplay=2, order=None):
             Order of dimensions, where last `ndisplay` will be
             rendered in canvas.
         """
+        slice_input = self._make_slice_input(point, ndisplay, order)
+
+        if self._slice_input == slice_input:
+            return
```

Comment:
I haven't thought this through entirely, since we won't be removing slice state on the layer for the first of async slicing.

But a natural place feels like the `LayerSlicer` because that's the thing that deals with requests, so it could cache the last (or last few) request-response mappings.

The tricky thing about moving the slice state from the layer is that invalidating the cache becomes harder because state updates (e.g. a change in transforms) to the layer may now need to invalidate the cache in the `LayerSlicer`.